third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -216,15 +220,19 @@ impl LazyLoadedFunction {
                         captured_layouts,
                     },
             } => {
-                let fun = Self::resolve::<RTTCheck>(
-                    module_storage,
-                    layout_converter,
+                let fun = loader.load_closure(
                     gas_meter,
                     traversal_context,
                     module_id,
                     fun_id,
                     ty_args,
-                    *mask,
+                )?;
+                Self::check_layouts_compatible::<RTTCheck>(
```

Comment:
As said elsewhere, I think we should remove this check. It adds unnecessary complexity.